packages/calypso-e2e/src/lib/pages/editor-page.ts

```diff
@@ -642,11 +631,14 @@ export class EditorPage {
 		}
 
 		// Resolve the promises.
-		const [ response ] = await Promise.all( actionsArray );
-
-		if ( ! response ) {
-			throw new Error( 'No response received from `publish` method.' );
-		}
+		const [ response ] = await Promise.all( [
+			// First URL matches Atomic requests while the second matches Simple requests.
+			Promise.race( [
+				this.page.waitForResponse( /v2\/(posts|pages)\/[\d]+/ ),
+				this.page.waitForResponse( /.*v2\/sites\/[\d]+\/(posts|pages)\/[\d]+.*/ ),
+			] ),
+			actionsArray,
```

Comment:
**Suggestion (blocking)**: I believe `actionsArray` here needs to be spread, right? We want to wait for all the items in that array, so we need to bring those items to the top level array in the `Promise.all` call.

e.g. `...actionsArray`

Reply:
Interesting how it seemed to work just fine without spreading...I thought multiple arrays inside a `Promise.all` would get treated as top level. 

In any case, made the change in e97b1e2bfe604ae56f47d0a725e2c9e6a9f39d99/